### ISSUE TITLE
there are few bugs if I am not wrong in the subtotal of the invoice

### ISSUE THREAD

**Issue by kishanmodi** [johnuberbacher/invoice-generator#1]
**When I checked the functionality of your app while I was trying to replicate your project, I encountered a few bugs.**

1. Subtotal is not calculating the remaining items other than the first one
![image](https://user-images.githubusercontent.com/45737372/185612958-b3fffaaa-1777-41e8-8570-10623ecb977e.png)

2. Showing wrong precision
![image](https://user-images.githubusercontent.com/45737372/185613052-d7d5a1b7-f30f-4494-8137-766a1e05efac.png)

3. Subtotal is not updated in the preview invoice as well.
 ![image](https://user-images.githubusercontent.com/45737372/185613139-2e6fc62b-6ee7-472d-bf0e-c9cbccb58d18.png)

I may be wrong as I am a beginner in react. Do check out these issues.

**Comment by amarcin3** [comment]
I found a solution to all problems. All bugs here are very easy to fix, but hard to find.
Solution for 1. and 3. In InvoiceForm.js around line 80 (mine says 84 because i have changed the code a little, you can also search for it using ctrl+f). you need to add another `phraseFloat()` because (i think, i am also not an expert) `.toFixed()` changes it back to string (and adding strings together is a mess) 
![git13](https://user-images.githubusercontent.com/80335507/193883882-457d3165-2a97-45b4-9e83-38eff22e8472.png)

Solution for 2. I am no longer sure what fixed it in my code, but I think it could have been one of `toFixed(2)` around 85 line and below (see image)
![git2](https://user-images.githubusercontent.com/80335507/193886180-fe6758e8-77e5-4017-9cb6-bed0c7ccc43e.png)
You can also add toString like on the image since the `taxAmmount` and `discountAmmount` are string values.

**Comment by kishanmodi** [comment]
@amarcin3 Back When I was learning, I created the same project. if you would like to check here's the link [Invoice Generator](https://github.com/kishanmodi/invoice-generator).

**Comment by amarcin3** [comment]
@kishanmodi Well, you have made some things work, but you also broke it a bit. When you delete the row above it breaks all the rows below. I know it was like a "one day project", and It won't be used as an actual invoice maker for any company, so it's fine.

**Comment by Vyomrana02** [comment]
Still there was error in subtotal calculation, Can @johnuberbacher u assign me this issue?

**Comment by rahulS-repo** [comment]
Just add these lines in InvoiceForm component and it will fix all issue

```
 handleCalculateTotal() {
    var itemList = this.state.items;
    var subTotal = 0;
    // items.map(function (items) {
    //   subTotal = parseFloat(
    //     subTotal + parseFloat(items.price).toFixed(2) * parseInt(items.quantity)
    //   ).toFixed(2);
    // });
    itemList.map(
      (itemDetail) =>
        (subTotal +=
          parseFloat(itemDetail.price).toFixed(2) *
          parseInt(itemDetail.quantity))
    );

```
i am also not an expert😅

**Comment by AbhaySingh15** [comment]
Yeah can confirm this is a bug, idk why the project creator used map function instead of forEach to calculate the subtotal, and also overall the function is kind of a mess. here is the updated code to fix the issue:

```java
handleCalculateTotal() {
    var items = this.state.items;
    var subTotal = 0;

    // this is a bug, it concatenates all the calculated values as a string instead
    // of adding them
    // items.map(function(items) {
    //   subTotal = parseFloat(subTotal + (parseFloat(items.price).toFixed(2) * parseInt(items.quantity))).toFixed(2)
    // });

    // This is the corrected code for subTotal
    items.forEach(function (item) {
      subTotal += parseFloat(item.price) * parseInt(item.quantity);
    });

    // Convert the accumulated subtotal to a string with two decimal places
   const subTotalString = subTotal.toFixed(2);
    console.log(subTotal);

    this.setState({
      subTotal: subTotalString
    }, () => {
      this.setState({
        taxAmmount: ((subTotal) * (this.state.taxRate / 100)).toFixed(2)
      }, () => {
        this.setState({
          discountAmmount: ((subTotal) * (this.state.discountRate / 100)).toFixed(2)
        }, () => {
          this.setState({
            total: ((subTotal - this.state.discountAmmount) + parseFloat(this.state.taxAmmount)).toFixed(2)
          });
        });
      });
    });
  };
```
